### PR TITLE
chore: migrate to JUnit assertThrows

### PR DESCRIPTION
**Pull request by teleivo**
in preparation to the migration to JUnit 5. assertThrows will be the way
Exceptions have to be asserted

https://junit.org/junit5/docs/current/user-guide/#writing-tests-assertions

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/security/EnrollmentSecurityTest.java [hunk 1]
```diff
@@ -233,27 +238,16 @@ public void testUserWithDataReadWriteNoOrgUnit()
...
-            createEnrollment( programA.getUid(), maleB.getUid() ), ImportOptions.getDefaultImportOptions() )
-            .getStatus() );
-
-        assertEquals( ImportStatus.ERROR, enrollmentService.addEnrollment(
-            createEnrollment( programA.getUid(), femaleA.getUid() ), ImportOptions.getDefaultImportOptions() )
-            .getStatus() );
-
-        assertEquals( ImportStatus.ERROR, enrollmentService.addEnrollment(
-            createEnrollment( programA.getUid(), femaleB.getUid() ), ImportOptions.getDefaultImportOptions() )
-            .getStatus() );
```
Comment (jbee) [inline dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/security/EnrollmentSecurityTest.java:250]: Where did these go?

File: dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/security/EnrollmentSecurityTest.java [hunk 2]
```diff
@@ -268,23 +262,17 @@ public void testUserWithDataReadOrgUnit()
...
-            createEnrollment( programA.getUid(), femaleA.getUid() ), ImportOptions.getDefaultImportOptions() )
-            .getStatus() );
+        assertThrows( IllegalQueryException.class,
+            () -> enrollmentService.addEnrollment(
+                createEnrollment( programA.getUid(), maleB.getUid() ), ImportOptions.getDefaultImportOptions() )
+                .getStatus() );
 
-        assertEquals( ImportStatus.ERROR, enrollmentService.addEnrollment(
-            createEnrollment( programA.getUid(), femaleB.getUid() ), ImportOptions.getDefaultImportOptions() )
-            .getStatus() );
```
Comment (jbee) [inline dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/security/EnrollmentSecurityTest.java:281]: And these?

**Review by jbee — APPROVED** [review]
Great stuff. 

In the future feel free if you do this sort of refactoring to also factor out helper methods for reoccurring patterns. I think if we don't do these in the same step they get left behind and will not be done.

File: dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/security/EnrollmentSecurityTest.java (hunk 1 above)
Comment (teleivo) [reply dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/security/EnrollmentSecurityTest.java:250]: They were never called. I removed the `expected = SomeException.class` and ran these tests to figure out which of the calls actually threw the exception.

File: dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/security/EnrollmentSecurityTest.java (hunk 2 above)
Comment (teleivo) [reply dhis-2/dhis-services/dhis-service-dxf2/src/test/java/org/hisp/dhis/dxf2/events/security/EnrollmentSecurityTest.java:281]: same as above

**Comment by teleivo** [comment]
> In the future feel free if you do this sort of refactoring so also factor out helper methods for reoccurring patterns. I think if we don't do these in the same step they get left behind and will not be done.

What helpers do you mean for example 😄 ? I try to keep the PRs focused as I am already touching many files and have a list of PRs ahead of me. I rather get the changes in quickly, otherwise, I'll be in rebase hell. Looking forward to some backports 😋

**Review by enricocolasante — APPROVED** [review]

**Comment by jbee** [comment]
>  I try to keep the PRs focused as I am already touching many files and have a list of PRs ahead of me.

Way to go. :+1: 

What I mean is if there are 4-5 times same `assertThrows` with same method call body but different arguments you could go and make a helper method with the `assertThrows` line where just the 1-2 parameters get varied. This makes the diff stay basically the same but the code is also getting a bit cleaner.

**Pull request merged**